src/NuGet.Core/NuGet.Build.Tasks/GetRestoreProjectReferencesTask.cs

```diff
@@ -47,6 +47,10 @@ public override bool Execute()
 
             var entries = new List<ITaskItem>();
 
+            // Filter obvious duplicates without considering OS case sensitivity.
+            // This will be filtered further when creating the spec.
+            var seen = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
It could use that check, but I think this is better handled at a higher level when creating the dg file. This part is just a helper in MSBuild that creates items with extra metadata.